x-pack/plugins/security_solution/public/management/components/endpoint_response_actions_list/components/status_badge.tsx

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import React, { memo } from 'react';
+import { EuiBadge, type EuiBadgeProps } from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { useTestIdGenerator } from '../../../hooks/use_test_id_generator';
+import { type UX_MESSAGES } from '../translations';
+
+export const StatusBadge = memo(
+  ({
+    color,
+    status,
+  }: {
+    color: EuiBadgeProps['color'];
+    status: keyof typeof UX_MESSAGES['badge'];
```

Comment:
Yeah, I want the status type to a union here of the specific status names. The line of code where the prop is being passed is doing some string slicing to uppercase the first letter of the status and then type case it to `ResponseActionStatus` which is the same as this union here. 

I'll clean up the types so that they are consistent. 👍

Reply:
Okay, so I need the prop type here to be a `string` as it is already i18n-ed.

Reply:
ef49415f8532a5d60fff9f9fd1f3de00070f43d2